src/anemoi/graphs/create.py

```diff
@@ -77,9 +77,14 @@ def clean(self, graph: HeteroData) -> HeteroData:
 
     def save(self, graph: HeteroData) -> None:
         """Save the graph to the output path."""
-        if not os.path.exists(self.path) or self.overwrite:
+        if self.path is None:
+            LOGGER.info("No output path specified. The graph will not be saved.")
+        elif not self.path.exists() or self.overwrite:
+            self.path.parent.mkdir(parents=True, exist_ok=True)
             torch.save(graph, self.path)
-            self.print(f"Graph saved at {self.path}.")
+            LOGGER.info(f"Graph saved at {self.path}.")
+        else:
```

Comment:
This is somewhat duplicated from self.init(). It should probably be part of the same logic.